libraries/tools/kotlin-gradle-plugin-api/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinSourceSet.kt

```diff
@@ -10,6 +10,7 @@ import org.gradle.api.Named
 import org.gradle.api.file.SourceDirectorySet
 
 interface KotlinSourceSet : Named, HasKotlinDependencies {
+    val id: Long
```

Comment:
In a single project, source set names are unique. Using the `name` as a key would be sufficient. Why a separate `id` is needed?

Reply:
Well, I am not a domain expert, but I can easily change this to maps of String.

Reply:
Fixed